spec/models/miq_filter_spec.rb

```diff
@@ -2,9 +2,9 @@
   let(:ems)             { FactoryGirl.create(:ems_vmware, :name => 'ems') }
   let(:ems_folder_path) { "/belongsto/ExtManagementSystem|#{ems.name}" }
   let(:datacenter)      { FactoryGirl.create(:ems_folder, :name => "Datacenters") }
-  let(:mtc)             { FactoryGirl.create(:ems_folder, :name => "MTC", :is_datacenter => true) }
+  let(:mtc)             { FactoryGirl.create(:datacenter, :name => "MTC") }
   let(:ems_folder_path) { "/belongsto/ExtManagementSystem|#{ems.name}" }
-  let(:mtc_folder_path) { "#{ems_folder_path}/EmsFolder|#{datacenter.name}/EmsFolder|#{mtc.name}" }
+  let(:mtc_folder_path) { "#{ems_folder_path}/EmsFolder|#{datacenter.name}/Datacenter|#{mtc.name}" }
```

Comment:
This changes things...there might already be tags or values in the database that will require a data migration.  *Or* the code can use base_model or base_class, and this can stay as EmsFolder